native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/internal/NativeImageCommandLineProvider.java

```diff
@@ -119,13 +125,15 @@ public List<String> asArguments() {
         appendBooleanOption(cliArgs, options.getVerbose(), "--verbose");
         appendBooleanOption(cliArgs, options.getSharedLibrary(), "--shared");
         appendBooleanOption(cliArgs, options.getQuickBuild(), "-Ob");
-        appendBooleanOption(cliArgs, options.getRichOutput(), "-H:+BuildOutputColorful");
+        appendBooleanOption(cliArgs, options.getRichOutput(), majorJDKVersion >= 21 ? "--color" : "-H:+BuildOutputColorful");
```

Comment:
Note: this option is only used in the Gradle plugin. Since Maven doesn't need it, it also doesn't need to check `native-image --version` unconditionally.